src/main/java/roboy/dialog/states/expoStates/RoboyQAState.java

```diff
@@ -0,0 +1,218 @@
+package roboy.dialog.states.expoStates;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import roboy.context.Context;
+import roboy.context.contextObjects.IntentValue;
+import roboy.dialog.states.definitions.State;
+import roboy.dialog.states.definitions.StateParameters;
+import roboy.linguistics.Linguistics;
+import roboy.linguistics.Triple;
+import roboy.linguistics.sentenceanalysis.Interpretation;
+import roboy.memory.Neo4jProperty;
+import roboy.memory.Neo4jRelationship;
+import roboy.memory.nodes.Interlocutor;
+import roboy.memory.nodes.MemoryNodeModel;
+import roboy.memory.nodes.Roboy;
+import roboy.talk.PhraseCollection;
+import roboy.util.QAJsonParser;
+import roboy.util.RandomList;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static roboy.memory.Neo4jProperty.*;
+
+public class RoboyQAState extends State {
+    public final static String INTENTS_HISTORY_ID = "RQA";
+
+    private final String SELECTED_SKILLS = "skills";
+    private final String SELECTED_ABILITIES = "abilities";
+    private final String LEARN_ABOUT_PERSON = "newPerson";
```

Comment:
getRequiredTransitions() function should be implemented to detect missing definitions in the personality file and simplify debugging

also, this state is not modular because it assumes that the transitions point to specific states

Reply:
getRequiredTransitions() implemented